compilation error on slackware-current (linux) with 3.14.1

* SuperCollider version: 3.14.1
* Operating system: slackware-current (64 bit, linux)
* Other details: qt6: 6.10.1, pipewire: 1.4.9, gcc: 15.2.0, glibc: 2.42, 6.12.60 kernel


When compiling the editor, the build fails:  https://gist.github.com/garpu/870564adfacc083593f99a6769ec691e

I had no issues building 3.14.0. I've tried both with and without system boost, since that's given me problems in the past. (Same error both times.)  I believe 3.14.0 was compiled against qt6 6.9.1. (I don't have an easy way to roll this back.) 

Here's the config: https://gist.github.com/garpu/fcd8bc9ebc6ffe09db3b541922f00114

Problem in question seems to be e.g. line 701

https://github.com/supercollider/supercollider/blob/4073839222f7fd8456cb9c3714786c44247ca594/editors/sc-ide/widgets/code_editor/editor.cpp#L675-L703

```
/tmp/SBo/SuperCollider-3.14.1-Source/editors/sc-ide/widgets/code_editor/editor.cpp: In member function ‘virtual void ScIDE::GenericCodeEditor::mousePressEvent(QMouseEvent*)’:
/tmp/SBo/SuperCollider-3.14.1-Source/editors/sc-ide/widgets/code_editor/editor.cpp:701:46: error: no matching function for call to ‘QString::arg(Qt::KeyboardModifiers)’
  701 |                                          .arg(e->modifiers())
      |                                              ^
/tmp/SBo/SuperCollider-3.14.1-Source/editors/sc-ide/widgets/code_editor/editor.cpp:701:46: note: there are 4 candidates
/usr/include/qt6/QtCore/qstring.h:324:27: note: candidate 1: ‘template<class T, typename std::enable_if<conjunction_v<std::disjunction<std::is_integral<_Tp>, std::conjunction<std::is_enum<_Tp>, std::negation<q23::is_scoped_enum<E, is_enum_v<E> > > > >, std::negation<std::disjunction<std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, _Float16>, std::is_same<typename std::remove_cv<typename std::remove_reference<_Tp>::type>::type, qfloat16>, std::is_floating_point<_Tp> > >, std::negation<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv< <template-parameter-1-1> >::type> >, std::is_convertible<T, QAnyStringView> > > >, bool>::type <anonymous> > QString QString::arg(T, int, int, QChar) const’
 
```

Taking a look at the [`QString::arg`](https://doc.qt.io/qt-6/qstring.html#arg-2) documentation tells us regarding 6.10

> Note: In Qt versions prior to 6.10.1, this function accepted arguments of types that implicitly convert to integral types. This is no longer supported, except for (unscoped) enums, because it also accepted types convertible to floating-point types, losing precision when those were printed as integers. A backwards-compatible fix is to cast such types to a C++ type whose displayed form matches your intent (int, float, ...).

So we only have to be more precise, which probably does not hurt backwards compatibility, which is great.

Sounds like a good first issue for people who have Qt6.10 on their machine :)

--- a/editors/sc-ide/widgets/code_editor/editor.cpp
+++ b/editors/sc-ide/widgets/code_editor/editor.cpp
@@ -698,7 +698,7 @@ void GenericCodeEditor::mousePressEvent(QMouseEvent* e) {
                                          .arg(e->position().x())
                                          .arg(e->position().y())
 #endif
-                                         .arg(e->modifiers())
+                                         .arg(static_cast<Qt::KeyboardModifiers::Int>(e->modifiers()))
                                          .arg(button),
                                      true);
     }
@@ -732,7 +732,7 @@ void GenericCodeEditor::mouseDoubleClickEvent(QMouseEvent* e) {
                                          .arg(e->position().x())
                                          .arg(e->position().y())
 #endif
-                                         .arg(e->modifiers())
+                                         .arg(static_cast<Qt::KeyboardModifiers::Int>(e->modifiers()))
                                          .arg(button),
                                      true);
     }
@@ -766,7 +766,7 @@ void GenericCodeEditor::mouseReleaseEvent(QMouseEvent* e) {
                                          .arg(e->position().x())
                                          .arg(e->position().y())
 #endif
-                                         .arg(e->modifiers())
+                                         .arg(static_cast<Qt::KeyboardModifiers::Int>(e->modifiers()))
                                          .arg(button),
                                      true);
     }

